### ISSUE TITLE
Add a `getInputs` function

### ISSUE THREAD

**Issue by dariooddenino** [thomashoneyman/purescript-halogen-formless#19]
I'm listening to `Changed` messages from a form that's essentially a set of filters for a table. I want to update my `filter` section of the state, but right now there's no "easy" way to grab all the inputs values at once.

I came up with this solution using `heterogeneous`:

```
extractResultFromChanged
  :: forall xs a form fields r
   . RL.RowToList fields xs
  => UnwrapRecord xs fields a
  => Newtype (form Record F.FormField) { | fields }
  => { form :: form Record F.FormField | r }
  -> { | a }
extractResultFromChanged = F.unwrapRecord <<< unwrap <<< _.form

newtype GetProp (prop :: Symbol) = GetProp (SProxy prop)

instance getProp ::
  (IsSymbol prop, Row.Cons prop a rx r) =>
  Mapping (GetProp prop) { | r } a where
  mapping (GetProp prop) = Record.get prop

getInputs' :: forall rin rout.
  HMap (GetProp "input") { | rin } { | rout } =>
  { | rin } ->
  { | rout }
getInputs' = hmap (GetProp (SProxy :: SProxy "input")) 

getInputs
   :: forall a xs form fields rout r ys
    . Newtype (form Record F.FormField) { | fields }
   => RL.RowToList fields xs
   => UnwrapRecord xs fields a
   => RL.RowToList a ys
   => HMap (GetProp "input") { | a } { | rout }
   => MapRecordWithIndex ys (ConstMapping (GetProp "input")) a rout
   => { form :: form Record F.FormField | r }
   -> { | rout }
getInputs = getInputs' <<< extractResultFromChanged
```

I just noticed that this is not consistent with `getInput` which takes the form as argument (here I'm extracting it from the message's payload.

**Comment by thomashoneyman** [comment]
Thanks for opening this issue and describing what you're doing. It does seem useful to be able to recover a record of only inputs from a user `Form` type, as well as potentially the same for errors, results, outputs, touched, etc. I could easily see someone wanting to work with any of those records.

There is a module that already exports `getInput`, which expects a symbol proxy and a form to recover the input from, and it seems sensible to me to add related functions for `getInputs`, `getErrors`, and so on.

**Comment by thomashoneyman** [comment]
@dariooddenino If you would like to contribute those using `purescript-heterogeneous` via a pull request, please feel free to do so and link this issue; if you're busy at the moment, then I'll make the update by Monday.

**Comment by dariooddenino** [comment]
I'm not able to work on it before Monday/Tuesday I think. If it's still missing then I'll create a pull request :)

**Comment by thomashoneyman** [comment]
Closed in #22 & #33!
